core/tests/window/test_window.py

```diff
@@ -99,6 +99,29 @@ def test_set_app_with_content(window, app):
     assert content.app == app
 
 
+def test_set_app_with_content_at_instantiation(app):
+    """A window can be created with content"""
+    try:
+        window_no_content = toga.Window()
+        assert window_no_content.content is None
+    finally:
+        window_no_content.close()
+        window_no_content = None
```

Comment:
There's no need to test the "no content" option here - this is tested extensively by existing tests.